docs/book/expressions.md

```diff
@@ -227,6 +227,60 @@ askName
     @name = $name
 ```
 
+## DEPLOY Variables
+
+DEPLOY variables are a type of memory storage variable that you can define as
+part of your Litexa configuration to exist at compilation time. They can be
+used like the other types of variables, but can also be used in [`when`
+statements](/reference/#when) and [file exclusion statements](/reference/#exclude-file).
+
+To define DEPLOY variables, add a `DEPLOY` object in your deployment targets
+like so:
+
+```javascript
+const deploymentConfiguration = {
+  name: 'cats-in-space',
+  deployments: {
+    development: {
+      ... // other configuration
+      DEPLOY: { // your DEPLOY variables go in this object
+        debug: true,
+        mode: 'practice'
+      }
+    },
+    production: {
+      ... // other configuration
+      DEPLOY: {
+        mode: 'challenge'
+      }
+    },
+    ...
+```
+
+These variables will then be available for use in your Litexa code with their
+values dependent on which deployment target you specified in the litexa command
+(`litexa test`, `litexa deploy`, etc.).
```

Comment:
Might be worth adding a warning to specify that the `DEPLOY` object should only be used for properties, and not for methods (which would presumably just be trimmed by the JSON stringification).